portfolio/src/main/java/com/google/sps/servlets/Comment.java

```diff
@@ -0,0 +1,80 @@
+package com.google.sps.servlets;
+
+/**
+ * Encapsulated the possible values for reasons a user visited a site.
+ * Update this enum when adding or removing allowable Visit Type values to the
+ * Comments form.
+ *
+ * NONE represents a null response (i.e. user did not answer the question)
+ * All other responses correspond with one radio button on the form
+ */
+enum VisitType {
+  NONE, RECRUITING, PROJECT, TUTORING, CHAT
+}
+
+/**
+ * Encapsulates a user-entered comment and their personal information,
+ * including their first and last name, reason for visiting, and email address.
+ *
+ * Can only be initialized once.
+ *
+ * Setter functions included, but getter functions are not needed
+ * since GSON accesses private members directly.
+ */
+public final class Comment {
+  private String email;
+  private String firstName;
+  private String lastName;
+  private VisitType visitReason;
+  private String comment;
+
+  /**
+   * Initializes a comment. All values are directly passed to their respective fields,
+   * except for visitReason which requires additional logic stored in a setter method.
+   * @param email email address of site visitor
+   * @param firstName first name of site visitor
+   * @param lastName last name of site visitor
+   * @param visitReason reason for the visit's visit. Must correspond to a value in the VisitType enum
+   * @param comment User's comment after visiting site.
+   */
+  public Comment(String email, String firstName, String lastName, String visitReason,
+      String comment) {
+    this.email = email;
+    this.firstName = firstName;
+    this.lastName = lastName;
+    this.comment = comment;
+    setVisitType(visitReason);
+  }
+
+    /**
+     * Set the VisitType enum to the appropriate value based on the string recieved in the POST
+     * request. If the input is unexpected, throw an error Expected inputs are one of the following:
+     * "", "recruiting", "project", "tutoring", or "chat".
+     * @param visitReason string representation from POST request of the reason user visited.
+     * @throws IllegalArgumentException the visitReason is not one of the above expected inputs.
+     */
```

Comment:
Isn't the indentation a bit off?

Reply:
Yep it was off by 2 - thanks for the catch. Fixed